api/python/provisioner/commands/generate_roster.py

```diff
@@ -0,0 +1,139 @@
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+import logging
+from typing import Type
+from pathlib import Path
+
+from ..utils import dump_yaml
+from .. import (
+    inputs,
+    config,
+    values
+)
+from ..vendor import attr
+
+from ..pillar import PillarKey, PillarResolver
+
+from . import (
+    CommandParserFillerMixin
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+@attr.s(auto_attribs=True)
+class RunArgsGenerateRosterAttrs:
+    roster_path: str = attr.ib(
+        metadata={
+            inputs.METADATA_ARGPARSER: {
+                'help': "roster file path"
+            }
+        },
+        default=str(config.PRVSNR_FACTORY_PROFILE_DIR / 'srv/roster')
+    )
+    user: str = attr.ib(
+        metadata={
+            inputs.METADATA_ARGPARSER: {
+                'help': "user for salt-ssh"
+            }
+        },
+        default='root'
+    )
+    port: int = attr.ib(
+        metadata={
+            inputs.METADATA_ARGPARSER: {
+                'help': "port for salt-ssh"
+            }
+        },
+        converter=int,
+        default=22
+    )
+    priv_key: str = attr.ib(
+        metadata={
+            inputs.METADATA_ARGPARSER: {
+                'help': "private key path for salt-ssh"
+            }
+        },
+        default=str(config.PRVSNR_FACTORY_PROFILE_DIR / '.ssh/setup.id_rsa')
+    )
+
+
+@attr.s(auto_attribs=True)
+class GenerateRoster(CommandParserFillerMixin):
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+    _run_args_type = RunArgsGenerateRosterAttrs
+
+    @staticmethod
+    def _get_pillar_data(key):
+        """Retrieve pillar data."""
+        pillar_key = PillarKey(key)
+        pillar = PillarResolver(config.LOCAL_MINION).get([pillar_key])
+        pillar = next(iter(pillar.values()))
+        if not pillar[pillar_key] or pillar[pillar_key] is values.MISSED:
+            raise ValueError(f"value is not specified for {key}")
+        else:
+            return pillar[pillar_key]
+
+    @staticmethod
+    def _get_nodes_list():
+        """Retrieve node  list from pillar"""
+        nodes = []
+        for key in GenerateRoster._get_pillar_data("cluster"):
+            if 'srvnode' in key:
+                nodes.append(key)
+        return nodes
+
+    @staticmethod
+    def _get_node_host_map():
+        """Provide map of node and hostname"""
+        host_node_map = {}
+        for node in GenerateRoster._get_nodes_list():
+            host = GenerateRoster._get_pillar_data(
+                       f'cluster/{node}/hostname')
+            host_node_map.update({node: host})
+        return host_node_map
+
+    @staticmethod
+    def _prepare_roster(
```

Comment:
Should not be a static method as the call to this method would be rare. To allocate memory for this is unnecessarily overloading the resources.

Reply:
comment is addressed